### PR TITLE
Photo fix

### PR DESCRIPTION
**Pull request by RichardNutt**
see issue: #213

### Visible/Frontend Changes
- added a "has photo" checkbox when editing child; check it when a file is deposited onn the server

### Architectural/Backend Changes
- removed onPhotoError() method, since it just set "hasPhoto" to false, regardless of the photos existence

### CODE REVIEW

File: src/app/children/child-details/child-details.component.ts [hunk 1]
```diff
@@ -109,6 +109,8 @@ export class ChildDetailsComponent implements OnInit {
       dropoutDate:    [{value: this.child.dropoutDate,    disabled: !this.editing}],
       dropoutType:    [{value: this.child.dropoutType,    disabled: !this.editing}],
       dropoutRemarks: [{value: this.child.dropoutRemarks, disabled: !this.editing}],
+
+      hasPhoto: [this.child.hasPhoto]
```
Comment (sleidig) [inline src/app/children/child-details/child-details.component.ts]: should we maybe save the filename instead? That'd give us more flexibility

Comment (RichardNutt) [reply src/app/children/child-details/child-details.component.ts]: Done.

**Comment by sleidig** [comment]
@RichardNutt If my additions look good (and clear) to you, go ahead and press the merge button.

File: src/app/children/child.ts [hunk 2]
```diff
@@ -100,9 +108,9 @@ export class Child extends Entity {
   }
 
   public getPhoto() {
-    if (!this.photoFile) {
+    if (!this.photoFile || this.photoFile === '') {
```
Comment (RichardNutt) [inline src/app/children/child.ts]: empty string already evaluate to false

**Pull request merged**